src/parcsr_ls/par_relax.c

```diff
@@ -393,9 +393,10 @@ hypre_BoomerAMGRelax1GaussSeidel( hypre_ParCSRMatrix *A,
       num_sends = hypre_ParCSRCommPkgNumSends(comm_pkg);
       num_recvs = hypre_ParCSRCommPkgNumRecvs(comm_pkg);
 
-      v_buf_data = hypre_CTAlloc(HYPRE_Real, hypre_ParCSRCommPkgSendMapStart(comm_pkg, num_sends),
+      v_buf_data = hypre_CTAlloc(HYPRE_Complex,
+                                 hypre_ParCSRCommPkgSendMapStart(comm_pkg, num_sends),
                                  HYPRE_MEMORY_HOST);
-      v_ext_data = hypre_CTAlloc(HYPRE_Real, num_cols_offd, HYPRE_MEMORY_HOST);
+      v_ext_data = hypre_CTAlloc(HYPRE_Complex, num_cols_offd, HYPRE_MEMORY_HOST);
```

Comment:
The main reason for this change is because `v_ext_data` is declared as `HYPRE_Complex`, see https://github.com/hypre-space/hypre/blob/e042698bc5a1cc6b44274ce419335bd526753b59/src/parcsr_ls/par_relax.c#L379

Now, regarding the second question: we use `HYPRE_Complex` for storing the data array of our vectors and matrices, so I think it's correct to use `HYPRE_Complex` for other data arrays like `v_ext_data` to make it consistent. If we decide to use `HYPRE_Real` for matrix and vector data instead, then `v_ext_data` should be `HYPRE_Real`. At the end of the day, `HYPRE_Real` is the same as `HYPRE_Complex` in the way hypre is usually built. 

Side question not for this PR: should we have something like `HYPRE_Scalar` to minimize confusion? I like this idea, however the number of changes would be large.